gerbil.evaluate/src/main/java/org/aksw/gerbil/evaluate/impl/ROCCurve.java

```diff
@@ -160,7 +162,16 @@ public double calcualteAUC() {
             pointB = points.get(i);
             // calculate area under the points (rectangle)
             if (pointB.x != pointA.x) {
-                aup = pointA.y * (pointB.x - pointA.x);
+                // if the two points are a step to the right
+                if (pointB.y == pointA.y) {
+                    aup = pointA.y * (pointB.x - pointA.x);
+                } else {
+                    // this is a diagonal
+                    // rectangle "under B"
+                    aup = pointB.y * (pointB.x - pointA.x);
+                    // triangle from B to A
+                    aup += 0.5 * (pointA.y - pointB.y) * (pointB.x - pointA.x);
```

Comment:
why `pointA.y - pointB.y` 
Is `pointB` not the one which is greater here?  
(see upper comment)

Reply:
Wow... :thinking: 
This second error solved the problem of the first error because it adds a negative value to the `aup` representing the small triangle that was added in the first step but is not part of the AUC... :wink: